### PR TITLE
Added TestModal Component to Admin Test page

### PR DESCRIPTION
**Pull request by kneyzberg**
### Problem 
Before creating the Artist Carousel, we need to implement a Modal from react-modal library to create a full page overlay that can be opened and closed. 

### Solution 
- installed react-modal library
- installed types for some of our react libraries
- created TestModal Component
- added TestModal component to admin/test page

### CODE REVIEW

**Review by bunnymatic — APPROVED** [review]

File: frontend/javascript/react/components/test_modal.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React, { FC, useState } from "react";
+import Modal from 'react-modal';
+
+const TestModal: FC = () => {
+  const [modalOpen, setModalOpen] = useState(false);
```
Comment (bunnymatic) [inline frontend/javascript/react/components/test_modal.tsx:5]: In MAU we have a `useModalState` hook that wraps up all this useState stuff.  it's handy and a good example of how to build new hooks - cuz it's so teeny.  Once this is merged I can add that in - or if you want to copy it over from MAU it's in reactjs/useModalState.ts.  Also noticed right there is a `useCarouselState` hook that might also be useful.  feel free to steal from there and copy them over if you wanna.

Comment (kneyzberg) [reply frontend/javascript/react/components/test_modal.tsx:5]: Sounds good, I can port it over!

**Review by bunnymatic — APPROVED** [review]

**Review by bunnymatic — APPROVED** [review]

**Comment by bunnymatic** [comment]
Sorry about the merge thing - I updated a pile of packages.   :(

**Comment by bunnymatic** [comment]
I might just resolve these and add a commit here cuz I feel bad about it :D

**Pull request merged**